tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -18,6 +19,31 @@ include(`platform/intel/'PLATFORM`.m4')
 
 DEBUG_START
 
+# Configure demux
+# name, pipeline_id, routing_matrix_rows
+define(matrix1, `ROUTE_MATRIX(3,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+
+define(matrix2, `ROUTE_MATRIX(4,
+			     `BITS_TO_BYTE(1, 0, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 1, 0 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 1 ,0 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,1 ,0 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,1 ,0 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,1 ,0 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,1 ,0)',
+			     `BITS_TO_BYTE(0, 0, 0 ,0 ,0 ,0 ,0 ,1)')')
+
+# name, num_streams, route_matrix list
+MUXDEMUX_CONFIG(demux_priv, 2, LIST(`	', `matrix1,', `matrix2'))
```

Comment:
This is only for 2 rt1308 codecs case. So you will need
```patch
diff --git a/tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4 b/tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4
index f42bf5101..c5f68bf5b 100644
--- a/tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4
+++ b/tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4
@@ -19,6 +19,8 @@ include(`platform/intel/'PLATFORM`.m4')
 
 DEBUG_START
 
+ifdef(`MONO', `',
+`
 # Configure demux
 # name, pipeline_id, routing_matrix_rows
 define(matrix1, `ROUTE_MATRIX(3,
@@ -43,6 +45,7 @@ define(matrix2, `ROUTE_MATRIX(4,
 
 # name, num_streams, route_matrix list
 MUXDEMUX_CONFIG(demux_priv, 2, LIST(`	', `matrix1,', `matrix2'))
+')
 
 #
 # Define the pipelines
```

Reply:
@bardliao I'm having trouble to compile to file with mono flag... shouldn't the demux pipe addition be also behind MONO?

Reply:
@bardliao oh that was actually fixed in master, had old version